third-party/realdds/include/realdds/dds-option.h

```diff
@@ -20,37 +22,38 @@ struct dds_option_range
     float default_value = 0.0f;
 };
 
+class dds_stream_base;
+
 class dds_option
 {
     std::string _name;
     float _value = 0.0f;
     dds_option_range _range;
     std::string _description;
 
-    std::string _owner_name; //To identify device or stream having this option
+    dds_option( nlohmann::json const & );
 
-    dds_option( nlohmann::json const & j, const std::string & owner_name );
+    friend class dds_stream_base;
+    std::weak_ptr< dds_stream_base > _stream;
```

Comment:
We have talked about device level options and even general options, we did not want to include a stream concept for `dds_option`.

Reply:
Yes, so if it has a stream, it now has a `stream-name`, and if it's a device-level option, it doesn't. There's no other options right now... having the extra concept of an "owner" just made no sense once I remove the `owner_name` from the constructor. I think it's better understood this way, and it mirrors the way we implemented the `dds_stream_profile` and its `stream()` association.